firestore/integration_test_internal/src/validation_test.cc

```diff
@@ -120,49 +364,24 @@ class ValidationTest : public FirestoreIntegrationTest {
     DocumentSnapshot snapshot = ReadDocument(document);
 
     // snapshot paths
-    try {
-      snapshot.Get(path);
-      FAIL() << "should throw exception";
-    } catch (const std::invalid_argument& exception) {
-      EXPECT_EQ(reason, exception.what());
-    }
+    EXPECT_ERROR(snapshot.Get(path), reason);
 
     // Query filter / order fields
     CollectionReference collection = Collection();
     // WhereLessThan(), etc. omitted for brevity since the code path is
     // trivially shared.
-    try {
-      collection.WhereEqualTo(path, FieldValue::Integer(1));
-      FAIL() << "should throw exception" << path;
-    } catch (const std::invalid_argument& exception) {
-      EXPECT_EQ(reason, exception.what());
-    }
-    try {
-      collection.OrderBy(path);
-      FAIL() << "should throw exception";
-    } catch (const std::invalid_argument& exception) {
-      EXPECT_EQ(reason, exception.what());
-    }
+    EXPECT_ERROR(collection.WhereEqualTo(path, FieldValue::Integer(1)), reason);
+    EXPECT_ERROR(collection.OrderBy(path), reason);
 
     // update() paths.
-    try {
-      document.Update(MapFieldValue{{path, FieldValue::Integer(1)}});
-      FAIL() << "should throw exception";
-    } catch (const std::invalid_argument& exception) {
-      // TODO(b/171990785): Unify Android and C++ validation error messages.
-      EXPECT_THAT(
-          exception.what(),
-          AnyOf(
-              // When validated by iOS or common C++ code
-              StrEq(reason),
-              // When validated by Android Java code
-              StrEq("Use FieldPath.of() for field names containing '~*/[]'.")));
-    }
+    EXPECT_ERROR_EITHER(
+        document.Update(MapFieldValue{{path, FieldValue::Integer(1)}}), reason,
+        // TODO(b/171990785): Unify Android and C++ validation error messages.
+        // When validated by Android Java code
```

Comment:
This comment `When validated by Android Java code` doesn't make sense to me. Is it a copy-paste error? If not, could you re-write it as a full English sentence?

Reply:
Done. I think the original idea was something like "The message below is produced when validation is performed by the Android SDK", hopefully it reads a little better now.